File: crates/aegis-language/tests/corpora/python/modern_syntax.py
# modern_syntax.py — parser-tolerance corpus: modern Python with no tracked
# destructive or execution-sink calls.
#
# Expected: zero operations AND zero parse errors. This proves the pinned
# tree-sitter-python 0.25.0 grammar parses current Python cleanly and the
# adapter does not false-positive on modern constructs. Covers: `from
# __future__` annotations, walrus `:=`, positional-only `/`, keyword-only `*`,
# type hints, f-string `=` debug, `match`/`case`, decorators, and exception
# groups (`except*`, Python 3.11+).
from __future__ import annotations

import functools


def first(items: list[int], /, *, fallback: int = 0) -> int:
    # Walrus assignment in a condition.
    if (n := len(items)) > 0:
        return n
    # Structural pattern matching (Python 3.10+).
    match fallback:
        case 0:
            return -1
        case int() as v if v > 0:
            return v
        case _:
            return fallback


class Cached:
    # Decorator with a keyword argument.
    @functools.lru_cache(maxsize=4)
    def name(self) -> str:
        # f-string `=` debug (Python 3.8+).
        ident = "aegis"
        return f"{ident=}"


def grouped() -> None:
    # Exception groups (Python 3.11+).
    try:
        raise ValueError("x")
    except* (ValueError, OSError) as group:  # noqa: B904
        for exc in group.exceptions:
            print(exc)

File: crates/aegis-language/tests/corpora/javascript/modern_syntax.js
// Corpus: modern syntax (plan Iteration 7, ADR-022 §3).
//
// Proves the pinned tree-sitter-javascript 0.25.0 grammar parses current
// JavaScript cleanly and that the adapter surfaces no false operations over
// modern syntax constructs. Covers optional chaining, nullish coalescing,
// logical assignment, class fields / private fields / private methods /
// getters, async/await, arrow functions, template literals with and without
// interpolation, destructuring with defaults/rest, spread in call and array,
// exponentiation, numeric separators, BigInt, and optional catch binding.
//
// Every call site here is on a NON-tracked identifier (`fetch`, `fn`, `go`,
// `console`) or an optional-chaining member access, so none matches a tracked
// destructive/execution-sink API spelling — the corpus must yield zero ops.

const x = data?.items?.[0]?.name;
const y = data ?? "default";
config ||= {};
config.timeout ??= 5;

class Counter {
  #priv = 0;
  static count = 0;
  get value() {
    return this.#priv;
  }
  #bump() {
    this.#priv += 1;
    return this.#priv;
  }
}

async function go() {
  const r = await fetch(url);
  return r;
}
const inc = (n) => n + 1;

const msg = `hello ${name}`;
const lit = `plain`;

const [a, ...rest] = arr;
const {p, q = 2} = obj;
fn(...args);
const big = 1_000_000n;
const exp = 2 ** 10;

try {
  f();
} catch {}